pkg/proxy/logger/epinfo.go

```diff
@@ -45,9 +45,9 @@ type EndpointUpdater interface {
 	// for a new observed flow with the given characteristics.
 	UpdateProxyStatistics(l4Protocol string, port uint16, ingress, request bool, verdict accesslog.FlowVerdict)
 
-	// OnDNSPolicyUpdateLocked is called when the Endpoint's DNS policy has been updated.
+	// OnDNSPolicyUpdateForce is called when the Endpoint's DNS policy has been updated.
 	// 'rules' is a fresh copy of the DNS rules passed to the callee.
-	OnDNSPolicyUpdateLocked(rules restore.DNSRules)
+	OnDNSPolicyUpdateForce(rules restore.DNSRules)
```

Comment:
Updated as mentioned in similar comment.

Reply:
I think we still need to review all usage of this function to ensure that they're properly locking the endpoint when they update the DNS rules. This doesn't need to be done for this PR, but it could lead to strange behaviour in rare cases.